Build failure: opensc

## Steps To Reproduce

Steps to reproduce the behavior:

1. `nix-shell -p opensc`

## Build log

<details>

<summary>Build Log</summary>

```
fetching github input 'github:NixOS/nixpkgs/nixpkgs-unstable'
this derivation will be built:
  /nix/store/y898h1g31qmy2ya6ab8m6pbla6qi21x5-opensc-0.26.0.drv
building '/nix/store/y898h1g31qmy2ya6ab8m6pbla6qi21x5-opensc-0.26.0.drv'...
Running phase: unpackPhase
unpacking source archive /nix/store/vm9a72v1wyrhb63k9a2fbk712gwxd67l-source
source root is source
Running phase: patchPhase
Running phase: autoreconfPhase
autoreconf: export WARNINGS=
autoreconf: Entering directory '.'
autoreconf: configure.ac: not using Gettext
autoreconf: running: aclocal --force -I m4
autoreconf: configure.ac: tracing
autoreconf: running: libtoolize --copy --force
libtoolize: putting auxiliary files in AC_CONFIG_AUX_DIR, '.'.
libtoolize: copying file './ltmain.sh'
libtoolize: putting macros in AC_CONFIG_MACRO_DIRS, 'm4'.
libtoolize: copying file 'm4/libtool.m4'
libtoolize: copying file 'm4/ltoptions.m4'
libtoolize: copying file 'm4/ltsugar.m4'
libtoolize: copying file 'm4/ltversion.m4'
libtoolize: copying file 'm4/lt~obsolete.m4'
libtoolize: Remember to add 'LT_INIT' to configure.ac.
autoreconf: configure.ac: not using Intltool
autoreconf: configure.ac: not using Gtkdoc
autoreconf: running: aclocal --force -I m4
autoreconf: running: /nix/store/9q25vcr3f3qadw9sq2mslqzcgllw57hl-autoconf-2.72/bin/autoconf --force
autoreconf: running: /nix/store/9q25vcr3f3qadw9sq2mslqzcgllw57hl-autoconf-2.72/bin/autoheader --force
autoreconf: running: automake --add-missing --copy --force-missing
configure.ac:60: installing './compile'
configure.ac:59: installing './config.guess'
configure.ac:59: installing './config.sub'
configure.ac:31: installing './install-sh'
configure.ac:31: installing './missing'
src/common/Makefile.am: installing './depcomp'
parallel-tests: installing './test-driver'
autoreconf: './install-sh' is updated
autoreconf: Leaving directory '.'
Running phase: updateAutotoolsGnuConfigScriptsPhase
Updating Autotools / GNU config script to a newer upstream version: ./config.sub
Updating Autotools / GNU config script to a newer upstream version: ./config.guess
Running phase: configurePhase
fixing libtool script ./ltmain.sh
./configure
patching script interpreter paths in ./configure
./configure: interpreter directive changed from "#! /bin/sh" to "/nix/store/sk8z3kdy6iaky6sg972k6fkh3iw2rn3d-bash-5.2p37/bin/sh"
configure flags: --disable-static --disable-dependency-tracking --prefix=/nix/store/w4isnhp4q47xgyh5xdfwxcdabkwi0qa0-opensc-0.26.0 --enable-zlib --enable-readline --enable-openssl --enable-pcsc --enable-sm --enable-man --enable-doc --localstatedir=/var --sysconfdir=/etc --with-xsl-stylesheetsdir=/nix/store/nyi0wbv194s5cmc2pivnalwz13zvkvss-docbook-xsl-nons-1.79.2/xml/xsl/docbook --with-pcsc-provider=/nix/store/0z3fb8w573sw17h4mwmzyjak5yr49qrl-PCSC-11.0/Library/Frameworks/PCSC.framework/PCSC
checking for a BSD-compatible install... /nix/store/5g6np23z9q7vwwp1s5pxkgn8f8wrmihh-coreutils-9.5/bin/install -c
checking whether build environment is sane... yes
checking for a race-free mkdir -p... /nix/store/5g6np23z9q7vwwp1s5pxkgn8f8wrmihh-coreutils-9.5/bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking whether make supports nested variables... (cached) yes
checking build system type... aarch64-apple-darwin24.1.0
checking host system type... aarch64-apple-darwin24.1.0
checking for gcc... clang
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether the compiler supports GNU C... yes
checking whether clang accepts -g... yes
checking for clang option to enable C11 features... none needed
checking whether clang understands -c and -o together... yes
checking whether make supports the include directive... yes (GNU style)
checking dependency style of clang... none
checking whether the compiler supports GNU C++... yes
checking whether clang++ accepts -g... yes
checking for clang++ option to enable C++11 features... none needed
checking dependency style of clang++... none
checking for gcc... no
checking for objcc... no
checking for objc... no
checking for cc... cc
checking whether the compiler supports GNU Objective C... yes
checking whether cc accepts -g... yes
checking dependency style of cc... none
checking pkg-config is at least version 0.9.0... yes
checking for stdio.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for strings.h... yes
checking for sys/stat.h... yes
checking for sys/types.h... yes
checking for unistd.h... yes
checking whether byte ordering is bigendian... no
checking whether to build with code coverage support... no
checking whether C compiler accepts -Wunknown-warning-option... yes
checking whether C compiler accepts -Wshorten-64-to-32... yes
checking for valgrind... no
checking how to run the C preprocessor... clang -E
checking whether ln -s works... yes
checking for a sed that does not truncate output... /nix/store/9m372k9wy0f9aj54zhvsr65w2p9ndwnj-gnused-4.9/bin/sed
checking whether make sets $(MAKE)... (cached) yes
checking how to print strings... printf
checking for a sed that does not truncate output... (cached) /nix/store/9m372k9wy0f9aj54zhvsr65w2p9ndwnj-gnused-4.9/bin/sed
checking for grep that handles long lines and -e... /nix/store/ay0r3wrwbv5rq3mg3vhla045yjr49gka-gnugrep-3.11/bin/grep
checking for egrep... /nix/store/ay0r3wrwbv5rq3mg3vhla045yjr49gka-gnugrep-3.11/bin/grep -E
checking for fgrep... /nix/store/ay0r3wrwbv5rq3mg3vhla045yjr49gka-gnugrep-3.11/bin/grep -F
checking for ld used by clang... ld
checking if the linker (ld) is GNU ld... no
checking for BSD- or MS-compatible name lister (nm)... nm
checking the name lister (nm) interface... BSD nm
checking the maximum length of command line arguments... 786432
checking how to convert aarch64-apple-darwin24.1.0 file names to aarch64-apple-darwin24.1.0 format... func_convert_file_noop
checking how to convert aarch64-apple-darwin24.1.0 file names to toolchain format... func_convert_file_noop
checking for ld option to reload object files... -r
checking for file... file
checking for objdump... objdump
checking how to recognize dependent libraries... (cached) pass_all
checking for dlltool... no
checking how to associate runtime and link libraries... printf %s\n
checking for archiver @FILE support... @
checking for strip... strip
checking for ranlib... ranlib
checking command to parse nm output from clang object... ok
checking for sysroot... no
checking for a working dd... /nix/store/5g6np23z9q7vwwp1s5pxkgn8f8wrmihh-coreutils-9.5/bin/dd
checking how to truncate binary pipes... /nix/store/5g6np23z9q7vwwp1s5pxkgn8f8wrmihh-coreutils-9.5/bin/dd bs=4096 count=1
checking for mt... no
checking if : is a manifest tool... no
checking for dsymutil... dsymutil
checking for nmedit... no
checking for lipo... lipo
checking for otool... otool
checking for otool64... no
checking for -single_module linker flag... yes
checking for -exported_symbols_list linker flag... yes
checking for -force_load linker flag... yes
checking for dlfcn.h... yes
checking for objdir... .libs
checking if clang supports -fno-rtti -fno-exceptions... yes
checking for clang option to produce PIC... -fno-common -DPIC
checking if clang PIC flag -fno-common -DPIC works... yes
checking if clang static flag -static works... no
checking if clang supports -c -o file.o... yes
checking if clang supports -c -o file.o... (cached) yes
checking whether the clang linker (ld) supports shared libraries... yes
checking dynamic linker characteristics... darwin24.1.0 dyld
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... yes
checking whether to build static libraries... no
checking how to run the C++ preprocessor... clang++ -E
checking for ld used by clang++... ld
checking if the linker (ld) is GNU ld... no
checking whether the clang++ linker (ld) supports shared libraries... yes
checking for clang++ option to produce PIC... -fno-common -DPIC
checking if clang++ PIC flag -fno-common -DPIC works... yes
checking if clang++ static flag -static works... no
checking if clang++ supports -c -o file.o... yes
checking if clang++ supports -c -o file.o... (cached) yes
checking whether the clang++ linker (ld) supports shared libraries... yes
checking dynamic linker characteristics... darwin24.1.0 dyld
checking how to hardcode library paths into programs... immediate
checking for windres... no
checking for xsltproc... xsltproc
checking for git... no
checking xsl-stylesheets... /nix/store/nyi0wbv194s5cmc2pivnalwz13zvkvss-docbook-xsl-nons-1.79.2/xml/xsl/docbook
checking git checkout... no
checking for inline... inline
checking for sys/wait.h that is POSIX.1 compatible... yes
checking whether to enable assertions... yes
checking for errno.h... yes
checking for fcntl.h... yes
checking for stdlib.h... (cached) yes
checking for inttypes.h... (cached) yes
checking for string.h... (cached) yes
checking for strings.h... (cached) yes
checking for sys/time.h... yes
checking for unistd.h... (cached) yes
checking for sys/mman.h... yes
checking for sys/endian.h... no
checking for endian.h... no
checking for an ANSI C-conforming const... yes
checking for uid_t... yes
checking for gid_t... yes
checking for size_t... yes
checking for error_at_line... no
checking whether lstat correctly handles trailing slash... no
checking whether stat accepts an empty string... no
checking for vprintf... yes
checking for getpass... yes
checking for gettimeofday... yes
checking for getline... yes
checking for memset... yes
checking for mkdir... yes
checking for strdup... yes
checking for strerror... yes
checking for memset_s... yes
checking for explicit_bzero... no
checking for strnlen... yes
checking for sigaction... yes
checking compiler support for __builtin_*_overflow()... yes
checking for clang options needed to detect all undeclared functions... none needed
checking whether strlcpy is declared... yes
checking whether strlcat is declared... yes
checking size of void *... 8
checking for socket in -lsocket... no
checking for dlopen in -ldl... yes
checking for egrep -e... (cached) /nix/store/ay0r3wrwbv5rq3mg3vhla045yjr49gka-gnugrep-3.11/bin/grep -E
checking whether clang is Clang... yes
checking whether pthreads work with "-pthread" and "-lpthread"... yes
checking whether Clang needs flag to prevent "argument unused" warning when linking with -pthread... no
checking for joinable pthread attribute... PTHREAD_CREATE_JOINABLE
checking whether more special flags are required for pthreads... no
checking for PTHREAD_PRIO_INHERIT... yes
checking for cmocka >= 1.0.1... no
checking for setjmp.h... yes
checking for cmocka.h... no
checking for inflate in -lz... yes
checking for zlib.h... yes
checking for readline in -lreadline... yes
checking for readline/readline.h... yes
checking for libcrypto >= 1.1.1... yes
configure: WARNING: libeac not found by pkg-config
checking for eac/eac.h... no
configure: WARNING: OpenPACE headers not found
checking for EAC_CTX_init_pace... configure: WARNING: Cannot link against libeac
checking for EAC_OBJ_nid2obj... no
Package libeac was not found in the pkg-config search path.
Perhaps you should add the directory containing `libeac.pc'
to the PKG_CONFIG_PATH environment variable
No package 'libeac' found
configure: WARNING: use --enable-cvcdir=DIR
Package libeac was not found in the pkg-config search path.
Perhaps you should add the directory containing `libeac.pc'
to the PKG_CONFIG_PATH environment variable
No package 'libeac' found
Package libeac was not found in the pkg-config search path.
Perhaps you should add the directory containing `libeac.pc'
to the PKG_CONFIG_PATH environment variable
No package 'libeac' found
configure: WARNING: use --enable-x509dir=DIR
checking for winscard.h... no
configure: error: winscard.h is required for pcsc
error: builder for '/nix/store/y898h1g31qmy2ya6ab8m6pbla6qi21x5-opensc-0.26.0.drv' failed with exit code 1;
       last 25 log lines:
       > checking for zlib.h... yes
       > checking for readline in -lreadline... yes
       > checking for readline/readline.h... yes
       > checking for libcrypto >= 1.1.1... yes
       > configure: WARNING: libeac not found by pkg-config
       > checking for eac/eac.h... no
       > configure: WARNING: OpenPACE headers not found
       > checking for EAC_CTX_init_pace... configure: WARNING: Cannot link against libeac
       > checking for EAC_OBJ_nid2obj... no
       > Package libeac was not found in the pkg-config search path.
       > Perhaps you should add the directory containing `libeac.pc'
       > to the PKG_CONFIG_PATH environment variable
       > No package 'libeac' found
       > configure: WARNING: use --enable-cvcdir=DIR
       > Package libeac was not found in the pkg-config search path.
       > Perhaps you should add the directory containing `libeac.pc'
       > to the PKG_CONFIG_PATH environment variable
       > No package 'libeac' found
       > Package libeac was not found in the pkg-config search path.
       > Perhaps you should add the directory containing `libeac.pc'
       > to the PKG_CONFIG_PATH environment variable
       > No package 'libeac' found
       > configure: WARNING: use --enable-x509dir=DIR
       > checking for winscard.h... no
       > configure: error: winscard.h is required for pcsc
       For full logs, run 'nix log /nix/store/y898h1g31qmy2ya6ab8m6pbla6qi21x5-opensc-0.26.0.drv'.
```

</details>

## Metadata

fetching github input 'github:NixOS/nixpkgs/nixpkgs-unstable'
 - system: `"aarch64-darwin"`
 - host os: `Darwin 24.1.0, macOS 15.1`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Lix, like Nix) 2.91.1
System type: aarch64-darwin
Additional system types: x86_64-darwin
Features: gc, signed-caches
System configuration file: /etc/nix/nix.conf
User configuration files: /Users/mumu/.config/nix/nix.conf:/etc/xdg/nix/nix.conf
Store directory: /nix/store
State directory: /nix/var/nix
Data directory: /nix/store/1phn7b1smb3g9jqz7v39q13wlyji2msz-lix-2.91.1/share`
 - nixpkgs: `/nix/store/n1g84klfb0h3bpwyvc59lcy5ca58h36w-source`

## Notify maintainers

@michaeladler

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/op/opensc/package.nix
+++ b/pkgs/by-name/op/opensc/package.nix
@@ -73,7 +73,12 @@ stdenv.mkDerivation rec {
     ) "XSLTPROC=${buildPackages.libxslt}/bin/xsltproc")
   ];
 
-  PCSC_CFLAGS = lib.optionalString withApplePCSC "-I${darwin.apple_sdk.frameworks.PCSC}/Library/Frameworks/PCSC.framework/Headers";
+  PCSC_CFLAGS = lib.concatStringsSep " " (
+    lib.optionals withApplePCSC [
+      "-I${darwin.apple_sdk.frameworks.PCSC}/Library/Frameworks/PCSC.framework/Headers"
+      "-I${lib.getDev pcsclite}/include/PCSC"
+    ]
+  );
 
   installFlags = [
     "sysconfdir=$(out)/etc"

